frontend/src/screens/dashboard/docsqa/main/context.tsx

```diff
@@ -0,0 +1,303 @@
+
+import { DocsQAContextType, SelectedRetrieverType } from './types'
+
+interface DocsQAProviderProps {
+  children: ReactNode
+}
+
+const defaultRetrieverConfig = `{
+  "search_type": "similarity",
+  "k": 20,
+  "fetch_k": 20,
+  "filter": {}
+}`
+
+const defaultModelConfig = `{
+  "parameters": {
+    "temperature": 0.1
+  }
+}`
+
+const defaultPrompt =
+  'Answer the question based only on the following context:\nContext: {context} \nQuestion: {question}'
+
+const DocsQAContext = createContext<DocsQAContextType | undefined>(undefined)
+
+export const DocsQAProvider: React.FC<DocsQAProviderProps> = ({ children }) => {
+  const [selectedQueryModel, setSelectedQueryModel] = React.useState('')
+  const [selectedCollection, setSelectedCollection] = useState('')
+  const [selectedQueryController, setSelectedQueryController] = useState('')
+  const [selectedRetriever, setSelectedRetriever] = useState<
+    SelectedRetrieverType | undefined
+  >()
+  const [prompt, setPrompt] = useState('')
+  const [isRunningPrompt, setIsRunningPrompt] = useState(false)
+  const [answer, setAnswer] = useState('')
+  const [sourceDocs, setSourceDocs] = useState<SourceDocs[]>([])
+  const [errorMessage, setErrorMessage] = useState(false)
+  const [modelConfig, setModelConfig] = useState(defaultModelConfig)
+  const [retrieverConfig, setRetrieverConfig] = useState(defaultRetrieverConfig)
+  const [promptTemplate, setPromptTemplate] = useState(defaultPrompt)
+
+  const [isInternetSearchEnabled, setIsInternetSearchEnabled] = useState(false)
+  const [isCreateApplicationModalOpen, setIsCreateApplicationModalOpen] =
+    useState(false)
+
+  const { data: collections, isLoading: isCollectionsLoading } =
+    useGetCollectionNamesQuery()
+  const { data: allEnabledModels } = useGetAllEnabledChatModelsQuery()
+  const { data: openapiSpecs } = useGetOpenapiSpecsQuery()
+  const [createApplication, { isLoading: isCreateApplicationLoading }] =
+    useCreateApplicationMutation()
+
+  const allQueryControllers = useMemo(() => {
+    if (!openapiSpecs?.paths) return []
+    return Object.keys(openapiSpecs?.paths)
+      .filter((path) => path.includes('/retrievers/'))
+      .map((str) => {
+        var parts = str.split('/')
+        return parts[2]
+      })
+  }, [openapiSpecs])
+
+  const allRetrieverOptions = useMemo(() => {
+    const queryControllerPath = `/retrievers/${selectedQueryController}/answer`
+    const examples =
+      openapiSpecs?.paths[queryControllerPath]?.post?.requestBody?.content?.[
+        'application/json'
+      ]?.examples
+    if (!examples) return []
+    return Object.entries(examples).map(([key, value]: [string, any]) => ({
+      key,
+      name: value.value.retriever_name,
+      summary: value.summary,
+      config: value.value.retriever_config,
+      promptTemplate: value.value.prompt_template ?? defaultPrompt,
+    }))
+  }, [selectedQueryController, openapiSpecs])
+
+  const handlePromptSubmit = async () => {
```

Comment:
Define submit handler where used, Are multiple components submitting the form?

Reply:
no no, i have changes this one and also that createApplication function